File: bookshop-front-angular/src/models/CartItem.ts
import { Book } from "./Book";

export class CartItem {
    public id: number = 0;
    public quantity: number = 0;
    public book: Book | null = null;

    constructor(id: number, quantity: number, book: any) {
        this.id = id;
        this.quantity = quantity;
        this.book = new Book(book);
    }
}

File: bookshop-back-api/bookshop.webapi/Models/CartFolder/CartItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace bookshop.webapi.Models.CartFolder
{
    public class CartItem
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public Book Book { get; set; }
        public Cart Cart { get; set; }
        public int Quantity { get; set; }
    }
}
